app/controllers/application_controller.rb

```diff
@@ -5,6 +5,147 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if logged_in?
+       redirect "/tweets"
+    else
+      erb :"/users/create_user"
+    end
+  end
+
+  post '/signup' do
+    if !params[:username].empty? && !params[:email].empty? &&!params[:password].empty?
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.